soroban-env-host/src/host_object.rs

```diff
@@ -1,16 +1,22 @@
 use super::{
+    host::metered_bigint::MeteredBigInt,
+    host::metered_map::MeteredOrdMap,
+    host::metered_vector::MeteredVector,
     weak_host::WeakHost,
     xdr::{self, ScObjectType},
     EnvVal, Object, RawVal,
 };
 
-use im_rc::{OrdMap, Vector};
-use num_bigint::BigInt;
-
 pub(crate) type HostObj = EnvVal<WeakHost, Object>;
 pub(crate) type HostVal = EnvVal<WeakHost, RawVal>;
-pub(crate) type HostMap = OrdMap<HostVal, HostVal>;
-pub(crate) type HostVec = Vector<HostVal>;
+pub(crate) type HostMap = MeteredOrdMap<HostVal, HostVal>;
+pub(crate) type HostVec = MeteredVector<HostVal>;
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub(crate) struct Sha256Hash([u8; 32]);
```

Comment:
I actually removed these yesterday -- they're redundant with XDR types that define hashes and public keys in terms of opaque arrays, there's really no need to convert to separate types here.

Reply:
Sorry about that, removed.